src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/validator/AbstractIdentifierValidator.java

```diff
@@ -1,75 +0,0 @@
-/**
- * The MIT License
- * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
- * Copyright (c) 2018 Estonian Information System Authority (RIA),
- * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
- * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-package org.niis.xroad.restapi.openapi.validator;
-
-import ee.ria.xroad.common.validation.EncodedIdentifierValidator;
-
-import lombok.Builder;
-import lombok.Data;
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.StringUtils;
-import org.springframework.validation.Errors;
-import org.springframework.validation.Validator;
-
-import java.util.Collection;
-import java.util.EnumSet;
-
-/**
- * Base class for validators that validate objects with fields that are used to create encoded identifiers
- */
-@Slf4j
-abstract class AbstractIdentifierValidator implements Validator {
```

Comment:
Removing this hides some validation error data. 

For example when adding a new subsystem `CS:ORG:1234:Sub%Sys\\t;em` would return the following validation error:
```
{
    "status": 400,
    "error": {
        "code": "validation_failure",
        "validation_errors": {
            "clientAdd.client.subsystemCode": [
                "IdentifierChars"
            ]
        }
    }
}
```

where the old validation would return a more specific validation error:
```
{
    "status": 400,
    "error": {
        "code": "validation_failure",
        "validation_errors": {
            "clientAdd.client.subsystemCode": [
                "NoPercents",
                "NoSemicolons",
                "NoBackslashes"
            ]
        }
    }
}
```

Reply:
This is true, it was a conscious design decision. If we wanted to keep that level of detail in the outcoming validation errors, we would need to implement separate validators for each (`NoPercentsValidator`, `NoSemicolonsValidator` etc.) instead of relying on a single validator `IdentifierCharsValidator` as we do now. I think from the end-users perspective this wouldn't add much since we can formulate the error messages in a generic manner e.g. "The field contains unallowed characters [%;\...]"

Reply:
It is probably a good compromise between codebase maintainability and user experience